src/Storages/Kafka/StorageKafka.cpp

```diff
@@ -846,6 +879,24 @@ void StorageKafka::updateConfiguration(cppkafka::Configuration & kafka_config)
     }
 }
 
+void StorageKafka::updateConsumerConfiguration(cppkafka::Configuration & kafka_config)
+{
+    const auto & config = getContext()->getConfigRef();
+    auto config_prefix = getConfigPrefix();
+    if (config.has(config_prefix))
+        loadConsumerConfig(kafka_config, config, config_prefix, topics);
+
+    loadTopicLegacyConfig(kafka_config, config, config_prefix, topics);
+}
+
+void StorageKafka::updateProducerConfiguration(cppkafka::Configuration & kafka_config)
+{
+    const auto & config = getContext()->getConfigRef();
+    auto config_prefix = getConfigPrefix();
+    if (config.has(config_prefix))
+        loadProducerConfig(kafka_config, config, config_prefix, topics);
```

Comment:
Silly question but why don't we `loadTopicLegacyConfig()` as in l. 889?

Reply:
I agreed to remove this legacy syntax at all and throw a warning if it is used.